src/shared/yagna/repository/proposal-repository.ts

```diff
@@ -1,15 +1,19 @@
-import { IProposalRepository, OfferProposal } from "../../../market/offer-proposal";
-import { MarketApi } from "ya-ts-client";
+import { OfferProposal } from "../../../market/proposal/offer-proposal";
+import { MarketApi, IdentityApi } from "ya-ts-client";
 import { Demand, GolemMarketError, MarketErrorCode } from "../../../market";
 import { CacheService } from "../../cache/CacheService";
+import { IProposalRepository } from "../../../market/proposal/types";
+import { MarketProposal } from "../../../market/proposal/market-proposal";
+import { OfferCounterProposal } from "../../../market/proposal/offer-counter-proposal";
 
 export class ProposalRepository implements IProposalRepository {
   constructor(
-    private readonly api: MarketApi.RequestorService,
-    private readonly cache: CacheService<OfferProposal>,
+    private readonly marketService: MarketApi.RequestorService,
+    private readonly identityService: IdentityApi.DefaultService,
```

Comment:
because dependency injection in our DI container is synchronous and we would have to do it only after the `.connect()` is executed, and this method (`getByDemandAndId`) seems to be nowhere used so it can probably be removed..